### ISSUE TITLE
Resource version enable/disable UI improvements

### ISSUE THREAD

**Issue by mmb** [concourse/concourse#4211]
In the resource version list UI, it's not entirely clear what the check mark does.

Recently we spent a while trying to figure out why a job wouldn't trigger and it turned out to be because the only known version had been inadvertently disabled, which grays it out and hides the checkmark.

Maybe this UI could be improved to indicate what this button does and to make it clearer when a resource is disabled.

![image](https://user-images.githubusercontent.com/24275/62336939-65063100-b487-11e9-8118-a1fa354f8331.png)

**Comment by vito** [comment]
Yeah having an empty box there is a little confusing with no other context. :thinking: 

The least we could do is add a tooltip, but we could maybe also add an icon of some sort there so it's not just an empty box.

**Comment by mohsen0** [comment]
@vito 
here is some question about this page, I can not find any related information about:
What makes a version to be enabled from start?
So I have a job, it has `trigger: true` so it will trigger and produce a new version of software each time a PR gets merged. on the other hand, I have another job that deploys the same service into staging and receives the output of the previous job (a semver version & docker shasum version& git sha version). this job has `trigger: true` enabled as well, but for the last job, I want to have the versions to be disabled so I can choose the version and at least control the version I don't want to deploy into production.

- have "Enable/disabling of versions" been documented somewhere?

**Comment by matthewpereira** [comment]
![image](https://user-images.githubusercontent.com/6372810/68211687-041a5780-ffa6-11e9-9e29-6a154f3972f2.png)

I made a few checkbox options.

I desk demoed them to a few folks in the office, and this was the winner so far:

![image](https://user-images.githubusercontent.com/6372810/68212150-f44f4300-ffa6-11e9-8a95-cd77ce176277.png)

![Frame 2(5)](https://user-images.githubusercontent.com/6372810/68212529-b272cc80-ffa7-11e9-8e27-db45c7045561.png)

Let me know if anyone wants a closer look at any of the other options and I can mock them out further. 

@Lindsayauchin @pivotal-jamie-klassen @zoetian 👀

**Comment by pivotal-jamie-klassen** [comment]
square-in-square looks a lot more natural than i expected!

**Comment by Lindsayauchin** [comment]
@matthewpereira @pivotal-jamie-klassen @zoetian 
I'm not sure about introducing rounded corners for the text box and the check mark. Looks like this icon is from a different set than the [Material Design icons](https://material.io/resources/icons/?search=check&icon=check_box&style=baseline) that we are using everywhere else in the app. For the resource icons we are using https://materialdesignicons.com/ so take a look in there. Can we just outline the square the check mark is already in? Doubling it up looks off balance. 

![Screen Shot 2019-11-05 at 9 42 46 AM](https://user-images.githubusercontent.com/27746614/68217472-a7706a00-ffb0-11e9-8259-5dbc50c2cad9.png)

**Comment by Lindsayauchin** [comment]
Adding the outline is looking pretty busy on the page. 

A strike through the box without adding outline logic is maybe the easiest route to indicate disabled state. 


@matthewpereira @pivotal-jamie-klassen @zoetian 

![Screen Shot 2019-11-05 at 10 51 26 AM](https://user-images.githubusercontent.com/27746614/68223223-65e4bc80-ffba-11e9-8039-688460c33304.png)


![Screen Shot 2019-11-05 at 10 53 01 AM](https://user-images.githubusercontent.com/27746614/68223254-75640580-ffba-11e9-9e26-9a16a8286a0e.png)

**Comment by Lindsayauchin** [comment]
@vito @matthewpereira @pivotal-jamie-klassen @zoetian 

How about a radio button to have a clearer metaphor to convey on/off/disabled?


![Screen Shot 2019-11-06 at 3 42 24 PM](https://user-images.githubusercontent.com/27746614/68336259-0e267e00-00ac-11ea-9fe7-f7fd83cb8331.png)

**Comment by matthewpereira** [comment]
Since radio buttons are used in situations where you can only select _one_ of several options (the options are mutually exclusive), I think we might be better off choosing one of the other options.

Likewise, a [-] in the context of a checkbox has some baggage because it normally denotes an indeterminate selection:
![image](https://user-images.githubusercontent.com/6372810/68424714-8d7e8500-0172-11ea-8966-ea3a050c233b.png)
...which you can change through javascript (like `somecheckbox.indeterminate = true;`).

It typically comes into play when you nest checkboxes, and want to show that a group is partially selected:
![image](https://css-tricks.com/wp-content/uploads/2011/08/indeterminatecheckboxes.png)

The [-] might not be as confusing as the radio button, (and I think it _looks_ pretty cool too) but we should probably get some feedback from a range of people just in case.

**Comment by Lindsayauchin** [comment]
@vito @mmb @pivotal-jamie-klassen @matthewpereira  

Brainstorming other options beyond the checkmark and the radio button alone we could combine a checkmark and the x with a toggle to show versions that are activated and deactivated in the list. 

<img width="737" alt="Screen Shot 2019-11-11 at 11 15 06 AM" src="https://user-images.githubusercontent.com/27746614/68602551-8d350100-0474-11ea-82c5-bb0bc0b2884d.png">

Thoughts?

**Comment by vito** [comment]
@Lindsayauchin What about a 🚫 icon that's dimmed/inactive normally and lit up red when the version is disabled? IMO that fits the usage pattern; disabling a resource version should be rare/emergent and behaves like a ban/block, i.e. a "red" state on the version. I think that's more suited to the workflow than showing all versions as "on" except for the ones which are turned "off".

**Comment by taylorsilva** [comment]
# Issue Bankruptcy

This issue is being closed as part of 2025's **Issue Bankruptcy** declaration. [See here for details](https://blog.concourse-ci.org/posts/2025-01-06-concourse-2024-review/)

If you think this issue still has merit then feel free to open a new issue and reference this one.
